doc/devices/dts/flash_partitions.rst

```diff
@@ -1,49 +1,228 @@
 .. _flash_partitions:
 
-Flash partitions nodes explanation
-##################################
+Device Tree Flash Partitions
+############################
 
-In each of the <boards>.dts files, the flash partitions layout can be described.
-It must be put inside the 'partitions' node in the 'flash0' node.
+Device tree can be used to describe a partition layout for any flash
+device in the system.
 
-Mcuboot related partitions
-**************************
+Two important uses for this mechanism are:
+
+#. To force the Zephyr image to be linked into a specific area on
+   Flash.
+
+   This is useful, for example, if the Zephyr image must be linked at
+   some offset from the flash device's start, to be loaded by a
+   bootloader at runtime.
+
+#. To generate compile-time definitions for the partition layout,
+   which can be shared by Zephyr subsystems and applications to
+   operate on specific areas in flash.
+
+   This is useful, for example, to create areas for storing file
+   systems or other persistent state.  These defines only describe the
+   boundaries of each partition. They don't, for example, initialize a
+   partition's flash contents with a file system.
+
+Partitions are generally managed using device tree overlays. Refer to
+:ref:`application_dt` for details on using overlay files.
+
+Defining Partitions
+*******************
+
+The partition layout for a flash device is described inside the
+``partitions`` child node of the flash device's node in the device
+tree.
+
+You can define partitions for any flash device on the system.
+
+Most SoCs supported by Zephyr which have flash support in device tree
```

Comment:
Too many whiches :)   How about:

    Most Zephyr-supported SoCs with flash support in device tree
    will define a label ``flash0``.   This label refers to the primary
    on-die flash programmed to run Zephyr. To generate partitions
    for this device, add the following snippet to a device tree overlay file: